### ISSUE TITLE
Improve Performance for higher dimensions

### ISSUE THREAD

**Issue by jeanbellynck** [jeanbellynck/Cayley-Graph-Generator#51]
The program becomes very slow when running in high dimensions. I guess this is due to the Barnes hut algorithm. 
Every cube has 2^n subcubes stored so the memory might be increasing exponentially with the dimension.

**Comment by p4jo** [comment]
Can the clustering be "wider" in the extra dimensions?

**Comment by p4jo** [comment]
Also, why not use System.Numerics.Vector<float>?

**Comment by jeanbellynck** [comment]
> Also, why not use System.Numerics.Vector<float>?

Good point actually. I only did a quick search and did not find any vector class.
I haven't closed this issue yet because I would like to replace my class with the vector class even if the performance gains will only be minor.
